pallets/dapp-staking/src/migration.rs

```diff
@@ -54,6 +54,92 @@ pub mod versioned_migrations {
             Pallet<T>,
             <T as frame_system::Config>::DbWeight,
         >;
+
+    /// Migration V8 to V9 wrapped in a [`frame_support::migrations::VersionedMigration`], ensuring
+    /// the migration is only performed when on-chain version is 8.
+    pub type V8ToV9<T, InitArgs> = frame_support::migrations::VersionedMigration<
+        8,
+        9,
+        v9::VersionMigrateV8ToV9<T, InitArgs>,
+        Pallet<T>,
+        <T as frame_system::Config>::DbWeight,
+    >;
+}
+
+/// Adaptation for the new `TierParams`, which include the arguments for 'number of slots' formula.
+mod v9 {
+    use super::*;
+    use frame_support::DefaultNoBound;
+
+    #[derive(
+        Encode,
+        Decode,
+        MaxEncodedLen,
+        RuntimeDebugNoBound,
+        PartialEqNoBound,
+        DefaultNoBound,
+        EqNoBound,
+        CloneNoBound,
+        TypeInfo,
+    )]
+    #[scale_info(skip_type_params(NT))]
+    pub struct TierParametersV8<NT: Get<u32>> {
+        /// Reward distribution per tier, in percentage.
+        /// First entry refers to the first tier, and so on.
+        /// The sum of all values must not exceed 100%.
+        /// In case it is less, portion of rewards will never be distributed.
+        pub(crate) reward_portion: BoundedVec<Permill, NT>,
+        /// Distribution of number of slots per tier, in percentage.
+        /// First entry refers to the first tier, and so on.
+        /// The sum of all values must not exceed 100%.
+        /// In case it is less, slot capacity will never be fully filled.
+        pub(crate) slot_distribution: BoundedVec<Permill, NT>,
+        /// Requirements for entry into each tier.
+        /// First entry refers to the first tier, and so on.
+        pub(crate) tier_thresholds: BoundedVec<TierThreshold, NT>,
+    }
+
+    pub struct VersionMigrateV8ToV9<T, InitArgs>(PhantomData<(T, InitArgs)>);
+
+    impl<T: Config, InitArgs: Get<(u64, u64)>> UncheckedOnRuntimeUpgrade
+        for VersionMigrateV8ToV9<T, InitArgs>
+    {
+        fn on_runtime_upgrade() -> Weight {
```

Comment:
Question: Is maintenance mode needed here?

Reply:
Not needed since we do the migration in a single block.
And this won't have immediate effect (unless it's the very same block at which tier reconfig should be done).